crates/dids/src/methods/dht/document_packet/verification_method.rs

```diff
@@ -0,0 +1,295 @@
+use std::{collections::HashMap, str::FromStr};
+
+use crate::document::VerificationMethod;
+
+use base64::{engine::general_purpose, Engine as _};
+use crypto::{ed25519::Ed25519, secp256k1::Secp256k1, Curve};
+use simple_dns::{
+    rdata::{RData, TXT},
+    Name, ResourceRecord,
+};
+
+use super::{
+    rdata_encoder::{get_rdata_txt_value, record_rdata_to_hash_map},
+    DocumentPacketError, DEFAULT_TTL,
+};
+
+#[derive(Debug, PartialEq)]
+struct VerificationMethodRdata {
+    pub t: String,
+    pub k: String,
+    pub c: Option<String>,
+    pub a: Option<String>,
+}
+
+impl TryFrom<HashMap<String, String>> for VerificationMethodRdata {
+    fn try_from(rdata_map: HashMap<String, String>) -> Result<Self, Self::Error> {
+        Ok(VerificationMethodRdata {
+            t: get_rdata_txt_value(&rdata_map, "t")?,
+            k: get_rdata_txt_value(&rdata_map, "k")?,
+            c: get_rdata_txt_value(&rdata_map, "c").ok(),
+            a: get_rdata_txt_value(&rdata_map, "a").ok(),
+        })
+    }
+
+    type Error = DocumentPacketError;
+}
+
+impl VerificationMethod {
+    pub fn to_resource_record(
+        &self,
+        did_uri: &str,
+        idx: u32,
+    ) -> Result<ResourceRecord<'static>, DocumentPacketError> {
```

Comment:
why does it have to be static?

Reply:
ah, that was a mistake. updated to remove the lifetime annotation.